R/plotScree.R

```diff
@@ -0,0 +1,169 @@
+#'
+#' @param x a
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-constructor]{TreeSummarizedExperiment}}
+#' or a vector of eigenvalues.
+#' 
+#' @param dimred \code{Character scalar} or \code{integer scalar}. Determines
+#' the reduced dimension to plot. This is used when x is a TreeSummarizedExperiment
+#' to extract the eigenvalues from \code{reducedDim(x, dimred)}.
+#' 
+#' @param cumulative \code{Logical scalar}. Whether to show cumulative explained 
+#' variance. (Default: \code{FALSE}).
+#' 
+#' @param names \code{Character vector}. Optional names for the components 
+#' that will be displayed on the x-axis. If not provided, the components 
+#' are labeled sequentially as 1, 2, 3, etc.
+#' 
+#' @param ... additional parameters for plotting
+#' \describe{
+#'   \item{\code{show.barplot}}{Logical scalar. Whether to show a barplot. 
+#'   (Default: \code{TRUE}).}
+#'   \item{\code{show.points}}{Logical scalar. Whether to show points. 
+#'   (Default: \code{TRUE}).}
+#'   \item{\code{show.line}}{Logical scalar. Whether to show a line connecting 
+#'   points. (Default: \code{TRUE}).}
+#'   \item{\code{show.labels}}{Logical scalar. Whether to show labels for each 
+#'   point. (Default: \code{FALSE}).}
+#' }
+#'
+#' @details
+#' \code{plotScree} creates a scree plot or eigenvalues plot, which is useful
+#' for visualizing the relative importance of components in dimensionality
+#' reduction techniques like PCA, RDA, or CCA. When the input is a
+#' TreeSummarizedExperiment, the function extracts eigenvalues from the specified
+#' reduced dimension slot. When the input is a vector, it directly uses these
+#' values as eigenvalues.
+#' 
+#' The plot can include a combination of barplot, points, connecting lines,
+#' and labels, which can be controlled using the \code{show.*} parameters.
+#' 
+#' An option to show cumulative explained variance is also available by setting
+#' \code{cumulative = TRUE}.
+#' 
+#' @return 
+#' A \code{ggplot2} object 
+#'
+#' @name plotScree
+#'
+#' @examples
+#' # Load necessary libraries
+#' library(ggplot2)
+#' 
+#' # Load dataset
+#' library(miaViz)
+#' data("enterotype", package = "mia")
+#' tse <- enterotype
+#'  
+#' # Run RDA and store results into TreeSE
+#' tse <- addRDA(
+#'     tse,
+#'     formula = assay ~ ClinicalStatus + Gender + Age,
+#'     FUN = getDissimilarity,
+#'     distance = "bray",
+#'     na.action = na.exclude
+#'     )
+#' 
+#' # Plot scree plot
+#' plotScree(tse, "RDA")
+#' 
+NULL
+
+#' @rdname plotScree
+#' @export
+setGeneric("plotScree", signature = c("x"),
+    function(x, ...) standardGeneric("plotScree"))
+
+#' @rdname plotScree
+#' @export
+setMethod("plotScree", signature = c(x = "SingleCellExperiment"),
+    function(x, dimred, cumulative = FALSE, ...) {
+        # Check if dimred exists
+        if (!dimred %in% reducedDimNames(x)) {
+            stop("'dimred' must specify a valid reducedDim.", call. = FALSE)
```

Comment:
We could also support index (that could be supported but not described in the documentation --> simpler when less options)

Reply:
I'm not sure I understand this. Do you mean subsetting the vector based on indices for example `indices = c(4,5)`?

Reply:
Sorry about unclear message, I meant `reducedDim(tse, 1)`, but I think you did it correctly